tsl/src/bgw_policy/job_api.c

```diff
@@ -55,6 +55,15 @@ job_add(PG_FUNCTION_ARGS)
 
 	TS_PREVENT_FUNC_IF_READ_ONLY();
 
+	if (!OidIsValid(proc))
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE), errmsg("function cannot be NULL")));
```

Comment:
It is allowed to provide a function or a procedure, thus it will be good to be clear in the error message. Saying `function` can confuse a user that a procedure is not expected.

Reply:
this code-path can be reached with non-NULL aswell so the error message is incorrect, probably should add a test for ```select add_job(0,'1day');```

Reply:
@k-rus I was following the format of other errors in this function that refers to a "function". I would have to update also those messages for consistency. That's OK with me, but I wanted to minimize changes.

Reply:
@erimatnor this PR contains changes to error message unrelated to this PR (removing hash in front of job number), so it is inline to improve other error messages as part of this PR.